src/main/java/com/umc/networkingService/domain/board/service/BoardServiceImpl.java

```diff
@@ -0,0 +1,232 @@
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class BoardServiceImpl implements BoardService {
+
+    private final BoardRepository boardRepository;
+    private final BoardFileService boardFileService;
+    private final BoardMapper boardMapper;
+    private final BoardHeartRepository boardHeartRepository;
+    private final BoardHeartMapper boardHeartMapper;
+
+
+    @Override
+    public BoardPagingResponse showBoards(Member member, HostType hostType, BoardType boardType, Pageable pageable) {
+        return boardMapper.toBoardPagingResponse(boardRepository.findAllBoards(member, hostType, boardType, pageable));
+    }
+
+    @Override
+    public BoardDetailResponse showBoardDetail(Member member, UUID boardId) {
+
+        Board board = loadEntity(boardId);
+
+        //조회수 증가
+        board.increaseHitCount();
+
+        //해당 게시글의 모든 첨부파일 url
+        List<String> boardFiles = boardFileService.findBoardFiles(board).stream()
+                .map(BoardFile::getUrl).toList();
+
+        return boardMapper.toBoardDetailResponse(board, boardFiles);
+
+    }
+
+    @Override
+    public BoardPagingResponse searchBoard(Member member, String keyword, Pageable pageable) {
+
+        return boardMapper.toBoardPagingResponse(boardRepository.findKeywordBoards(member, keyword, pageable));
+
+    }
+
+    @Override
+    @Transactional
+    public BoardIdResponse toggleBoardLike(Member member, UUID boardId) {
+        Board board = loadEntity(boardId);
+
+        //boardHeart에 없다면 새로 저장
+        BoardHeart boardHeart = boardHeartRepository.findByMemberAndBoard(member, board)
+                .orElseGet(() -> {
+                    BoardHeart newHeart = boardHeartMapper.toBoardHeartEntity(board, member);
+                    boardHeartRepository.save(newHeart);
+                    board.addBoardHeart(newHeart);
+                    return newHeart;
+                });
+
+        boardHeart.toggleHeart();
+        board.setHeartCount(boardHeart.isChecked());
+
+        return new BoardIdResponse(boardId);
+    }
+
+    @Override
+    @Transactional
+    public BoardIdResponse createBoard(Member member, BoardCreateRequest request, List<MultipartFile> files) {
+        //연합, 지부, 학교 타입
+        HostType hostType = request.getHostType();
+        BoardType boardType = request.getBoardType();
+
+        //boardType과 HostType에 따라 권한 판단
+        List<Semester> semesterPermission = checkPermission(member, hostType, boardType);
+
+        Board board = boardRepository.save(boardMapper.toEntity(member,
+                request.getTitle(),
+                request.getContent(),
+                request.getHostType(),
+                request.getBoardType(),
+                semesterPermission));
```

Comment:
BoardMapper내의 toEntity 함수의 파라미터를 dto 단위로 받는 것이 가독성과 유지 보수에 좋을 거 같아요!!!
```suggestion
        Board board = boardRepository.save(boardMapper.toEntity(member, request, semesterPermission));
```